### ISSUE TITLE
numfig feature improvement

### ISSUE THREAD

**Issue by shimizukawa** [sphinx-doc/sphinx#1586]
We've merged https://bitbucket.org/birkenfeld/sphinx/pull-request/296 for "Assign numbers to figures, tables and code-blocks" from [Takeshi Komiya](https://bitbucket.org/tk0miya). I have requests to improve the feature.

This feature assigns number to captions of code-block, figure, table with link target label as below:

```
.. _target:

.. code-block:: bash
   :caption: installation

   $ pip install Sphinx
```

When `numfig = True` in conf.py, this reST will be converted into html as:

```
<div id="id3" class="container">
  <div class="code-block-caption">
  List 1 installation
  </div>
  <div class="highlight-bash">
    <div class="highlight">
      <pre>
        $ pip install Sphinx
      </pre>
    </div>
  </div>
</div>
```

I think it would be better with:
1. `List 1 installation` would have permalink as like as section with using `config.html_add_permalinks` text.
2. `List 1 installation` would be wrapped with two `<span>` tag as: `<span class="numfig-number">List 1</span> <span class="caption-text">installation</span>` for css decoration. class names are just idea.

---
- Bitbucket: https://bitbucket.org/birkenfeld/sphinx/issue/1586
- Originally reported by: Takayuki Shimizukawa
- Originally created at: 2014-10-05T04:37:54.963

**Comment by shimizukawa** [comment]
_From Takeshi Komiya on 2014-10-05 12:04:20+00:00_

I send https://bitbucket.org/birkenfeld/sphinx/pull-request/304. I hope it satisfies you.

Thanks.

**Comment by shimizukawa** [comment]
_From Takayuki Shimizukawa on 2014-10-07 14:26:38+00:00_

https://bitbucket.org/birkenfeld/sphinx/pull-request/304 has been merged. Thanks!

**Comment by shimizukawa** [comment]
_From Andrea Cassioli on 2014-12-02 10:29:12+00:00_

Hi,
will it be possible in a similar way to have the number in the link instead of the caption?

**Comment by shimizukawa** [comment]
_From Takeshi Komiya on 2014-12-02 12:51:47+00:00_

Use `:numref:` instead of `:ref:`. It is already available at 1.3b1
